configs/encapsulate_in_http2_connect.yaml

```diff
@@ -26,10 +26,11 @@ static_resources:
           stat_prefix: tcp_stats
           cluster: "cluster_0"
           tunneling_config:
-            hostname: host.com
```

Comment:
sorry, why 10002 when it's connecting to port 10001?

Reply:
My idea was to show that the destination port can be different from the port used by the upstream proxy, but maybe using `443` makes more sense to be consistent with `terminate_connect.yaml`

Reply:
Sending  CONNECT foo.com:1234 to foo.com:1235 seems odd to me, but if you think it's worth explicitly testing for that how about a comment so other folks don't think it's just an off by one error :-)

Reply:
Actually, in this scenario, the CONNECT is sent to the upstream that is `127.0.0.1:10001` targeting `host.com:10002`. IMO it should be quite common to have the upstream proxy listening on a port and receiving CONNECT requests targeting a different one. What could be considered a bit odd maybe, is that the L2 proxy is receiving `CONNECT host.com:10002` and it is connecting instead to `www.google.com:443`.